samza-core/src/main/java/org/apache/samza/execution/JobNode.java

```diff
@@ -148,8 +148,6 @@ public JobConfig generateConfig(String executionPlanJson) {
     }
 
     if (!broadcasts.isEmpty()) {
-      // TODO: remove this once we support defining broadcast input stream in high-level
```

Comment:
I see the reason that you are appending the task.broadcast.inputs in inputDescriptor.toConfigure() now, since this line below assume that the descriptors don't tell us whether the stream is broadcast or not and only info is from the configure. Another way is that you can skip this part and just make changes to InputDescriptor to add the broadcast option. As long as my earlier comment on populating the broadcastStreams set in the ApplicationDescriptorImpl is done, we can get the broadcast streams from the ApplicationDescriptorImpl. In my PR, I have already changed the JobNode to include ApplicationDescriptorImpl as a parameter to the constructor.